pkg/tsdb/sqleng/sql_engine_test.go

```diff
@@ -9,317 +9,294 @@ import (
-
-			fixtures := make([]interface{}, 7)
-			fixtures[0] = tSeconds
-			fixtures[1] = &tSeconds
-			fixtures[2] = tMilliseconds
-			fixtures[3] = &tMilliseconds
-			fixtures[4] = tNanoSeconds
-			fixtures[5] = &tNanoSeconds
-			fixtures[6] = nilPointer
-
-			for i := range fixtures {
-				ConvertSqlTimeColumnToEpochMs(fixtures, i)
-			}
-
-			Convey("When converting them should return epoch time with millisecond precision ", func() {
-				So(fixtures[0].(float64), ShouldEqual, tMilliseconds)
-				So(fixtures[1].(float64), ShouldEqual, tMilliseconds)
-				So(fixtures[2].(float64), ShouldEqual, tMilliseconds)
-				So(fixtures[3].(float64), ShouldEqual, tMilliseconds)
-				So(fixtures[4].(float64), ShouldEqual, tMilliseconds)
-				So(fixtures[5].(float64), ShouldEqual, tMilliseconds)
-				So(fixtures[6], ShouldBeNil)
-			})
-		})
+	t.Run("Given row values with float32 as time columns", func(t *testing.T) {
+		tSeconds := float32(dt.Unix())
+		var nilInt *float32
 
-		Convey("Given row values with float32 as time columns", func() {
-			tSeconds := float32(dt.Unix())
-			var nilInt *float32
+		fixtures := make([]interface{}, 3)
+		fixtures[0] = tSeconds
+		fixtures[1] = &tSeconds
+		fixtures[2] = nilInt
 
-			fixtures := make([]interface{}, 3)
-			fixtures[0] = tSeconds
-			fixtures[1] = &tSeconds
-			fixtures[2] = nilInt
+		for i := range fixtures {
+			ConvertSqlTimeColumnToEpochMs(fixtures, i)
+		}
 
-			for i := range fixtures {
-				ConvertSqlTimeColumnToEpochMs(fixtures, i)
-			}
+		require.Equal(t, float64(tSeconds)*1e3, fixtures[0].(float64))
+		require.Equal(t, float64(tSeconds)*1e3, fixtures[1].(float64))
+		require.Nil(t, fixtures[2])
+	})
 
-			Convey("When converting them should return epoch time with millisecond precision ", func() {
-				So(fixtures[0].(float64), ShouldEqual, float32(dt.Unix()*1e3))
-				So(fixtures[1].(float64), ShouldEqual, float32(dt.Unix()*1e3))
-				So(fixtures[2], ShouldBeNil)
-			})
-		})
+	t.Run("Given row with value columns", func(t *testing.T) {
+		intValue := 1
+		int64Value := int64(1)
+		int32Value := int32(1)
+		int16Value := int16(1)
+		int8Value := int8(1)
+		float64Value := float64(1)
+		float32Value := float32(1)
+		uintValue := uint(1)
+		uint64Value := uint64(1)
+		uint32Value := uint32(1)
+		uint16Value := uint16(1)
+		uint8Value := uint8(1)
+
+		fixtures := []interface{}{
+			intValue,
+			&intValue,
+			int64Value,
+			&int64Value,
+			int32Value,
+			&int32Value,
+			int16Value,
+			&int16Value,
+			int8Value,
+			&int8Value,
+			float64Value,
+			&float64Value,
+			float32Value,
+			&float32Value,
+			uintValue,
+			&uintValue,
+			uint64Value,
+			&uint64Value,
+			uint32Value,
+			&uint32Value,
+			uint16Value,
+			&uint16Value,
+			uint8Value,
+			&uint8Value,
+		}
+
+		for _, f := range fixtures {
+			value, err := ConvertSqlValueColumnToFloat("col", f)
+			require.NoError(t, err)
+			require.True(t, value.Valid)
+			require.Equal(t, null.FloatFrom(1).Float64, value.Float64)
```

Comment:
```suggestion
			t.Run("", func(t *testing.T) {
				value, err := ConvertSqlValueColumnToFloat("col", f)
				require.NoError(t, err)
				assert.True(t, value.Valid)
				assert.Equal(t, null.FloatFrom(1).Float64, value.Float64)
			})
```

Reply:
Updated so that each fixture/test case is run as a subtest.